<launch>
<!-- Lidar node -->
  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
  <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
  <param name="serial_baudrate"     type="int"    value="115200"/><!--A1/A2 -->
  <!--param name="serial_baudrate"     type="int"    value="256000"--><!--A3 -->
  <param name="frame_id"            type="string" value="laser"/>
  <param name="inverted"            type="bool"   value="false"/>
  <param name="angle_compensate"    type="bool"   value="true"/>
  </node>

<!-- Odometry node
 <node pkg="odom_node_pkg" type="odom_node_type" name="odom_node" output="screen">
    <param name="odom_param" value="param_value" />
 </node>
-->
<!--static_transform_publisher x y z qx qy qz qw frame_id child_frame_id  period_in_ms  -->
<node pkg="tf" type="static_transform_publisher" name="base_link_to_lidar"        args="-0.012 0 0.0116 0 0 0 0 base_link base_laser 100" />
<node pkg="tf" type="static_transform_publisher" name="lidar_to_footprint"        args="0 0 -0.0191 0 0 0 0 base_link base_footprint 100" />
<node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0 0 0 0 0 0 map odom 30" />
<!--
<node pkg="tf" type="static_transform_publisher" name="imu_broadcaster" args="0 0.06 0.02 0 0 0 base_link imu 30" />
-->
 <!-- Transform_config for robot and lidar-->
<!-- <node pkg="robot_setup_tf" type="tf_broadcaster" name="robot_tf" output="screen"> -->
<!--    <param name="transform_configuration_param" value="param_value" /> -->


 <!-- Ros_serial com (com between rpi and teeny) -->
 <node pkg="rosserial_python" type="serial_node.py" name="rosserial_python" output="screen">
    <param name="port" value="/dev/ttyACM0" />
 </node>

 <node pkg="localization_data_pub" type="ekf_odom_pub" name="ekf_odom_pub">
 </node>

 <remap from="odom" to="odom_data_quat" />
 <remap from="imu_data" to="imu/data" />
 <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
   <param name="output_frame" value="odom"/>
   <param name="base_footprint_frame" value="base_footprint"/>
   <param name="freq" value="30.0"/>
   <param name="sensor_timeout" value="1.0"/>
   <param name="odom_used" value="true"/>
   <param name="imu_used" value="true"/>
   <param name="vo_used" value="false"/>
   <param name="gps_used" value="false"/>
   <param name="debug" value="false"/>
   <param name="self_diagnose" value="false"/>
 </node>

 <!--node pkg="rviz" type="rviz" name="rviz" args="-d /home/automaticaddison/catkin_ws/src/jetson_nano_bot/navigation_data_pub/maps/floorplan4.rviz">
</node> -->


 <arg name="map_file" default="$(find swarm_robot_nav)/maps/my_map.yaml"/>
-->
 -<node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)" />

 <include file="$(find amcl)/examples/amcl_diff.launch"/>

 <!-- Initial Pose and Goal Publisher -->
 <!-- Publish: /initialpose, /move_base_simple/goal -->
 <node pkg="rviz" type="rviz" name="rviz">
 </node>
 <!-- Subscribe: /initialpose, /move_base_simple/goal -->
 <!-- Publish: /initial_2d, /goal_2d -->
 <node pkg="localization_data_pub" type="rviz_click_to_2d" name="rviz_click_to_2d">
 </node>

</launch>
